package/MDAnalysis/analysis/rdf_s.py

```diff
@@ -0,0 +1,166 @@
+    step : int (optional)
+          The step size through the trajectory in frames (default is
+          every frame)
+
+    Example
+    -------
+    First create the :class:`InterRDF_s` object, by supplying one Universe
+    and two AtomGroup selection lists then use the :meth:`run` method ::
+
+      rdf = InterRDF_s(u, ags)
+      rdf.run()
+
+    Results are available through the :attr:`bins` and :attr:`rdf_s`
+    attributes::
+
+      plt.plot(rdf.bins, rdf.rdf_s[0])
+
+    To generate cdf, use the 'cdf_s' mehthod
+
+      rdf.cdf_s()
+
+    Results are available through the :attr:'cdf_s' attributes::
+      
+      plt.plot(rdf.bins, rdf.cdf_s[0])
+
+    .. versionadded:: 0.18.0
+
+    """
+    def __init__(self, u, ags,
+                 nbins=75, range=(0.0, 15.0), density=True, **kwargs):
+        super(InterRDF_s, self).__init__(u.universe.trajectory, **kwargs)
+
+        # List of AtomGroups 
+        self.ags = ags
+        
+        self.u = u
+        self._density = density
+
+        self.rdf_settings = {'bins': nbins,
+                             'range': range}
+
+    def _prepare(self):
+        # Empty list to store the RDF
+        count_list = []
+        count, edges = np.histogram([-1], **self.rdf_settings)
+        count_list = [np.zeros((ag1.n_atoms, ag2.n_atoms, len(count)), dtype=np.float64) 
+                         for ag1, ag2 in self.ags]
+
+        self.count = count_list
+        self.edges = edges
+        self.bins = 0.5 * (edges[:-1] + edges[1:])
+
+        # Need to know average volume
+        self.volume = 0.0
+
+
+    def _single_frame(self):
+        for i, (ag1, ag2) in enumerate(self.ags):
+            result=distances.distance_array(ag1.positions, ag2.positions,
+                                            box=self.u.dimensions)
+            for j in range(ag1.n_atoms):
+                for k in range(ag2.n_atoms):
+                    count = np.histogram(result[j, k], **self.rdf_settings)[0]
+                    self.count[i][j, k, :] += count
+
+        self.volume += self._ts.volume
+
+
+    def _conclude(self):
+        # Volume in each radial shell
+        vol = np.power(self.edges[1:], 3) - np.power(self.edges[:-1], 3)
+        vol *= 4/3.0 * np.pi
+
+        # Empty lists to restore indices, RDF
+        indices = []
+        rdf_s = []
+
+        for i, (ag1, ag2) in enumerate(self.ags):
+            # Number of each selection
+            nA = len(ag1)
+            nB = len(ag2)
+            N = nA * nB
+            indices.append([ag1.indices, ag2.indices])
+
+            # Average number density
+            box_vol = self.volume / self.n_frames
+            density = N / box_vol
+
+            if self._density:
+                rdf_s.append(self.count[i] / (density * vol * self.n_frames))
+            else:
+                rdf_s.append(self.count[i] / (vol * self.n_frames))
+
+        self.rdf_s = rdf_s
+        self.indices = indices
+
+    def get_cdf(self):
```

Comment:
add documentation
- What is being calculated?
- Where is it stored?
- What is returned?